components/entitlement/org.wso2.carbon.identity.entitlement/src/test/java/org/wso2/carbon/identity/entitlement/persistence/HybridConfigPersistenceManagerTest.java

```diff
@@ -0,0 +1,136 @@
+import org.wso2.carbon.identity.entitlement.internal.EntitlementServiceComponent;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+import static org.wso2.carbon.identity.entitlement.PDPConstants.Algorithms.*;
+
+/**
+ * This class tests the behavior of the HybridConfigPersistenceManager class.
+ */
+@WithCarbonHome
+@WithRegistry(injectToSingletons = {EntitlementServiceComponent.class})
+@WithRealmService(injectToSingletons = {EntitlementConfigHolder.class}, initUserStoreManager = true)
+@WithH2Database(jndiName = "jdbc/WSO2IdentityDB", files = {"dbscripts/h2.sql"})
+public class HybridConfigPersistenceManagerTest {
+
+    private HybridConfigPersistenceManager hybridConfigPersistenceManager;
+    private RegistryConfigPersistenceManager registryConfigPersistenceManager;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+
+        hybridConfigPersistenceManager = new HybridConfigPersistenceManager();
+        registryConfigPersistenceManager = new RegistryConfigPersistenceManager();
+    }
+
+    @Test
+    public void testGetDefaultGlobalPolicyAlgorithmName() throws Exception {
+
+        String globalPolicyAlgorithmName = registryConfigPersistenceManager.getGlobalPolicyAlgorithmName();
+        assertEquals(globalPolicyAlgorithmName, DENY_OVERRIDES);
+    }
+
+    @DataProvider
+    public Object[][] globalPolicyAlgorithmData() {
+
+        return new Object[][]{
+                {DENY_OVERRIDES},
+                {PERMIT_OVERRIDES},
+                {FIRST_APPLICABLE},
+                {ONLY_ONE_APPLICABLE},
+                {ORDERED_DENY_OVERRIDES},
+                {ORDERED_PERMIT_OVERRIDES}
+        };
+    }
+
+    @Test(dataProvider = "globalPolicyAlgorithmData")
+    public void testGetGlobalPolicyAlgorithmNameFromRegistry(String policyAlgorithm) throws Exception {
+
+        // Add the global policy combining algorithm.
+        boolean addResult = registryConfigPersistenceManager.addOrUpdateGlobalPolicyAlgorithm(policyAlgorithm);
+        assertFalse(addResult);
+
+        String policyAlgorithmFromRegistry = hybridConfigPersistenceManager.getGlobalPolicyAlgorithmName();
+        assertEquals(policyAlgorithmFromRegistry, policyAlgorithm);
+
+        registryConfigPersistenceManager.deleteGlobalPolicyAlgorithm();
+    }
+
+    @Test(dataProvider = "globalPolicyAlgorithmData")
+    public void testGetGlobalPolicyAlgorithmNameFromDatabase(String policyAlgorithm) throws Exception {
+
+        // Add the global policy combining algorithm.
+        boolean addResult = hybridConfigPersistenceManager.addOrUpdateGlobalPolicyAlgorithm(policyAlgorithm);
+        assertFalse(addResult);
+
+        String policyAlgorithmFromDatabase = hybridConfigPersistenceManager.getGlobalPolicyAlgorithmName();
+        assertEquals(policyAlgorithmFromDatabase, policyAlgorithm);
```

Comment:
This test does not guarantee that the execution has been done through the database impl. Lets fix that.

Reply:
Fixed in the latest changes.